documentation/gnmi/design/DASH.json

```diff
@@ -0,0 +1,724 @@
+            "encap": "vxlan",
+            "address_spaces": [
+                "10.1.0.0/16",
+                "10.2.0.0/16",
+                "2001:0123:1111:1000/64",
+                "2001:0123:1111:2000/64"
+            ]
+        }
+    }
+],
+
+//
+// active routing actions (active routing modules)
+// list of active modules impacts memory and needs to be configured as part of initial card config
+// Maximum 16 different routing actions (as each action ID will take 4 bits).
+//
+
+"routing-actions": [
+    "none",        // 0x0
+    "maprouting",  // 0x1
+    "direct",      // 0x2
+    "staticencap", // 0x3
+    "appliance",   // 0x4
+    "4to6",        // 0x5
+    "mapdecap",    // 0x6
+    "decap",       // 0x7
+    "drop"         // 0x8
+],
+
+//
+// Example memory how type of actions can be stored:
+//
+// -----------------------------------------------------------------------------------------------------------------------
+// | routing-type | order of actions | action-1 arguments | action-2 arguments | action-3 arguments | action-4 arguments |
+// -----------------------------------------------------------------------------------------------------------------------
+// Order of actions: 16-bit value that is split into 4x4 bits.
+// Example: 4to6 -> staticencap -> appliance -> none (will be represented as: 0x5340)
+// When packet hits the pipeline, it looks up LPM prefix, once the prefix is found, it gets routing type and persist arguments from LPM route
+// Then next stage takes routing type and looksup order ex. 0x5340 and persists all default arguments for stages
+// Then we go 4-times thru routing actions and apply applicable action to the packet based on carried arguments.
+// Important: "maprouting" has special implementation. If it is present (0x1000), then it will be only 1 action.
+// This action first does lookup to find the routing-type again based on prefix, and then performs the routing stages.
+//
+
+"routing-types-config": [
+    {
+        "direct": { 
+            "routing-type": "direct",
+            "actions": [ // up to 4 actions
+                {
+                    "action-type": "direct"
+                }
+            ]
+        }
+    },
+    {
+        "vpc": {
+            "routing-type": "vpc",
+            "actions": [ // ONLY 1 action here! More then 1 action is not allowed if action-type used is "maprouting"
+                {
+                    "action-type": "maprouting" // NOTE: "maprouting" will be the ONLY action. It cannot be used with other actions
+                }
+            ]
+        }
+    },
+    {
+        "vpc-direct": {
+            "routing-type": "vpc-direct",
+            "actions": [ // up to 3 actions
+                {
+                    "action-type": "staticencap",
+                    "encap-type": "vxlan"
+                }
+            ]
+        }
+    },
+    {
+        "drop": {
+            "routing-type": "drop",
+            "actions": [ // up to 3 actions
```

Comment:
For routing-type "drop", what are the other two action types that can be associated with “drop” action-type?